tracer/src/Datadog.Trace/PlatformHelpers/IContainerMetadata.cs

```diff
@@ -0,0 +1,50 @@
+// <copyright file="IContainerMetadata.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using Datadog.Trace.Agent;
+
+namespace Datadog.Trace.PlatformHelpers;
+
+internal interface IContainerMetadata
```

Comment:
Do we _need_ an interface? 🤔 If it's just for testing purposes, I think being able to provide `ContainerId`/`EntityId` in the constructor is as good as anything, seeing as it's ultimately pretty stateless? 

The main reason _not_ to use an interface here is that it's faster not to, because there's no devirtulaisation or anything required :)